src/App.jsx

```diff
@@ -1,34 +1,52 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
 import React from 'react';
-import './index.css';
-import { TopNavBar } from './components/TopNavBar/TopNavBar.jsx';
+import './index.scss';
+import { getFlights } from './services/flightsDataService'
+import { TopNavBar } from './components/TopNavBar/TopNavBar';
 import { SideNavBar } from './components/SideNavBar/SideNavBar';
 import { FlightsTable } from './components/FlightsTable/FlightsTable';
 
 export const App = () => {
-  const [flights, setFlights] = useState([
-    { id: 1, origin: 'DEN', destination: 'BOS'},
-    { id: 2, origin: 'DEN', destination: 'FLL'},
-    { id: 3, origin: 'DEN', destination: 'PVD'},
-    { id: 4, origin: 'ORD', destination: 'PVD'},
-    { id: 5, origin: 'PBI', destination: 'BOS'},
-  ]);
+  const [flights, setFlights] = useState(null);
+  const [origins, setOrigins] = useState([]);
+  const [originFilter, setOriginFilter] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
 
-  const origins = [...new Set(flights.map((flight) => flight.origin))];
+  useEffect(() => {
+    // Adding a timeout just to show off the loading effect 
+    setTimeout(() => {
+      const data = getFlights()
```

Comment:
The `const data = ` is pointless, unless for some reason you need to do something with the promise.

Also a bit of code smell having `data => ` in there, try to have unique variable names in similar blocks. But you can just remove the `const data = `

Reply:
Good catch. This is just me being spoiled and so used to async/await that I stumbled over the promise structure a bit. Fixed.

Reply:
@netia1128 BTW I generally try to use `async/await` inside my `useEffect`. No, you cannot do `useEffect(async () => { })` but you can at least do:

```
useEffect(() => {
    const fetch = async () => {
        try {
            await axios.get(......) .....
        }
        catch (ex) { ....... }
    }
    
    fetch();
}, []);
```

or 

```
useEffect(() => {
    (async () => {
        try {
            await axios.get(......) .....
        }
        catch (ex) { ....... }
    })();
}, []);
```

Reply:
My goodness. I spent so long trying to come up with the right syntax to do this with an async await as opposed to a promise. What you gave me above, plus the realization that I needed to put my setState functions into the async function itself was the secret sauce. Thank you so much 🙏